app/experimenter/experiments/api/v5/serializers.py

```diff
@@ -577,6 +577,38 @@ def _validate_duplicate_branch_names(self, data):
                         ],
                     }
                 )
+            try:
+                reference_branch = getattr(self.instance, "reference_branch")
+            except NimbusBranch.DoesNotExist:
+                reference_branch = None
+
+            try:
+                treatment_branches = getattr(self.instance, "treatment_branches")
+            except NimbusBranch.DoesNotExist:
+                treatment_branches = None
+
+            if self.instance and reference_branch and treatment_branches:
+                old_reference_branch_name = reference_branch.name
+
+                old_treatment_branch_names = [b.name for b in treatment_branches]
+
+                names_swapped = (
+                    old_reference_branch_name in treatment_branch_names
+                    and old_treatment_branch_names == [ref_branch_name]
+                )
+                if names_swapped:
+                    raise serializers.ValidationError(
+                        {
+                            "reference_branch": {
+                                "name": NimbusConstants.ERROR_BRANCH_SWAP
+                            },
+                            "treatment_branches": [
+                                {"name": NimbusConstants.ERROR_BRANCH_SWAP}
+                                for _ in data["treatment_branches"]
+                            ],
+                        }
+                    )
```

Comment:
This only catches the case of a single treatment swapping with control, but doesn't catch the cases of

1) two treatments swapping
2) swapping a treatment with reference when there's more than 1 treatment

See if you can find a more general way to check for swaps for any combination.